<a href="https://colab.research.google.com/github/mrivaldi/Project_PSDS/blob/main/Tugas_Akhir_A_Muhammad_Rivaldi_1800015094.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Google Play Store Apps**

berikut adalah data yang saya ambil dari kaggle, dimana data ini adalah data Google Play Store Aps. saya berniat mencari popularitar Apps berdasarkan category Apps pada dataset

## Pertanyaan Analisis

1. Category manakah yang paling sering diinstall?
2. Category manakah yang paling jarang diinstall?

## Data Preparation

untuk data prepraretion, saya menyiapkan data frame untuk jenis-jenis Category beserta jumlah Install-nya

In [9]:
import pandas as pd
import plotly.express as px

In [10]:
#import file csv dari github
df = pd.read_csv('https://raw.githubusercontent.com/mrivaldi/file/main/googleplaystore.csv')
df

App  ...         Android Ver
0         Photo Editor & Candy Camera & Grid & ScrapBook  ...        4.0.3 and up
1                                    Coloring book moana  ...        4.0.3 and up
2      U Launcher Lite – FREE Live Cool Themes, Hide ...  ...        4.0.3 and up
3                                  Sketch - Draw & Paint  ...          4.2 and up
4                  Pixel Draw - Number Art Coloring Book  ...          4.4 and up
...                                                  ...  ...                 ...
10836                                   Sya9a Maroc - FR  ...          4.1 and up
10837                   Fr. Mike Schmitz Audio Teachings  ...          4.1 and up
10838                             Parkinson Exercices FR  ...          2.2 and up
10839                      The SCP Foundation DB fr nn5n  ...  Varies with device
10840      iHoroscope - 2018 Daily Horoscope & Astrology  ...  Varies with device

[10841 rows x 13 columns]

In [11]:
#melihat nama-nama data serta jumlahnya yang ada pada colom Type
df.Installs.value_counts()

1,000,000+        1579
10,000,000+       1252
100,000+          1169
10,000+           1054
1,000+             907
5,000,000+         752
100+               719
500,000+           539
50,000+            479
5,000+             477
100,000,000+       409
10+                386
500+               330
50,000,000+        289
50+                205
5+                  82
500,000,000+        72
1+                  67
1,000,000,000+      58
0+                  14
0                    1
Free                 1
Name: Installs, dtype: int64

In [12]:
#menghilangkan tanda '+' dan ',' pada kolom Installs
#mengubah data 'Free' menjadi '0' pada kolom Installs
#mengkonversi type data pada kolom Installs  menjadi int
df['Installs'] = df['Installs'].str.replace('+', '')
df['Installs'] = df['Installs'].str.replace(',', '')
df['Installs'] = df['Installs'].str.replace('Free', '0')
df['Installs'] = df['Installs'].astype(int)
df

App  ...         Android Ver
0         Photo Editor & Candy Camera & Grid & ScrapBook  ...        4.0.3 and up
1                                    Coloring book moana  ...        4.0.3 and up
2      U Launcher Lite – FREE Live Cool Themes, Hide ...  ...        4.0.3 and up
3                                  Sketch - Draw & Paint  ...          4.2 and up
4                  Pixel Draw - Number Art Coloring Book  ...          4.4 and up
...                                                  ...  ...                 ...
10836                                   Sya9a Maroc - FR  ...          4.1 and up
10837                   Fr. Mike Schmitz Audio Teachings  ...          4.1 and up
10838                             Parkinson Exercices FR  ...          2.2 and up
10839                      The SCP Foundation DB fr nn5n  ...  Varies with device
10840      iHoroscope - 2018 Daily Horoscope & Astrology  ...  Varies with device

[10841 rows x 13 columns]

In [13]:
#membuat data List yang berisi semua Category yang ada pada Dataset
List_category = list(pd.unique(df['Category']))
List_category

['ART_AND_DESIGN',
 'AUTO_AND_VEHICLES',
 'BEAUTY',
 'BOOKS_AND_REFERENCE',
 'BUSINESS',
 'COMICS',
 'COMMUNICATION',
 'DATING',
 'EDUCATION',
 'ENTERTAINMENT',
 'EVENTS',
 'FINANCE',
 'FOOD_AND_DRINK',
 'HEALTH_AND_FITNESS',
 'HOUSE_AND_HOME',
 'LIBRARIES_AND_DEMO',
 'LIFESTYLE',
 'GAME',
 'FAMILY',
 'MEDICAL',
 'SOCIAL',
 'SHOPPING',
 'PHOTOGRAPHY',
 'SPORTS',
 'TRAVEL_AND_LOCAL',
 'TOOLS',
 'PERSONALIZATION',
 'PRODUCTIVITY',
 'PARENTING',
 'WEATHER',
 'VIDEO_PLAYERS',
 'NEWS_AND_MAGAZINES',
 'MAPS_AND_NAVIGATION',
 '1.9']

In [14]:
#membuat data List yang berisi Jumlah Installs Apps berdasarkan tiap Category
list_sum = []
for i in List_category:
  df_cat = df[df['Category'] == i]
  sum_ins = df_cat.Installs.sum()
  list_sum.append(sum_ins)

print(list_sum)

[124338100, 53130211, 27197050, 1921469576, 1001914865, 56086150, 32647276251, 264310807, 871452000, 2869160000, 15973161, 876648734, 273898751, 1583072512, 168712461, 62995910, 537643539, 35086024415, 10258263505, 53257437, 14069867902, 3247848785, 10088247655, 1751174498, 6868887146, 11452771915, 2325494782, 14176091369, 31521110, 426100520, 6222002720, 7496317760, 724281890, 0]


In [15]:
#membuat data frame dari jumlah install pada setiap category
df_install = pd.DataFrame({"Category":List_category,
              "Jumlah_Install":list_sum})
df_install.sort_values(['Jumlah_Install', 'Category'], ascending=[True, False], inplace=True)
df_install

Category  Jumlah_Install
33                  1.9               0
10               EVENTS        15973161
2                BEAUTY        27197050
28            PARENTING        31521110
1     AUTO_AND_VEHICLES        53130211
19              MEDICAL        53257437
5                COMICS        56086150
15   LIBRARIES_AND_DEMO        62995910
0        ART_AND_DESIGN       124338100
14       HOUSE_AND_HOME       168712461
7                DATING       264310807
12       FOOD_AND_DRINK       273898751
29              WEATHER       426100520
16            LIFESTYLE       537643539
32  MAPS_AND_NAVIGATION       724281890
8             EDUCATION       871452000
11              FINANCE       876648734
4              BUSINESS      1001914865
13   HEALTH_AND_FITNESS      1583072512
23               SPORTS      1751174498
3   BOOKS_AND_REFERENCE      1921469576
26      PERSONALIZATION      2325494782
9         ENTERTAINMENT      2869160000
21             SHOPPING      3247848785
30        VIDEO_PLAYERS      6222002720
24     TRAVEL_AND_LOCAL      6868887146
31   NEWS_AND_MAGAZINES      7496317760
22          PHOTOGRAPHY     10088247655
18               FAMILY     10258263505
25                TOOLS     11452771915
20               SOCIAL     14069867902
27         PRODUCTIVITY     14176091369
6         COMMUNICATION     32647276251
17                 GAME     35086024415

## Data Visualization

In [16]:
#Visualisasi Data dengan Membuat Plot Bar berdasarkan Dataframe df_install
gbr_3 = px.bar(df_install,
               x = 'Category',
               y = 'Jumlah_Install')

gbr_3.update_layout(
    title = 'Diagram Batang jumlah instal Apps Pada Google Play Store',
    title_font_color = 'blue',
    title_font_size = 30,

    xaxis = dict(
        title = 'Category',
        titlefont_size = 16,
        tickfont_size = 9),
    
    yaxis = dict(
        title = 'Jumlah Install',
        titlefont_size = 16,
        tickfont_size = 14))

gbr_3.show()

#**Jawaban Analisis Data**

1. Category Apps yang paling sering diinstall adalah Category 'Game'
2. Category Apps yang paling jarang diinstall adalah Category '1.9'